drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImpl.java

```diff
@@ -108,256 +108,266 @@ public DecisionTableXLSServiceImpl( @Named("ioStrategy") final IOService ioServi
-            if ( create ) {
-                ioService.createFile( nioPath );
+            validate(tempFile);
+
+            final org.uberfire.java.nio.file.Path nioPath = Paths.convert(resource);
+
+            ioService.startBatch(nioPath.getFileSystem());
+
+            if (create) {
+
+                tempFIS = new FileInputStream(tempFile);
+                ioService.write(nioPath,
+                                IOUtils.toByteArray(tempFIS),
+                                commentedOptionFactory.makeCommentedOption(comment,
+                                                                           sessionInfo.getIdentity(),
+                                                                           sessionInfo));
+            } else {
+
+                outputStream = ioService.newOutputStream(nioPath,
+                                                         commentedOptionFactory.makeCommentedOption(comment,
+                                                                                                    sessionInfo.getIdentity(),
+                                                                                                    sessionInfo));
+
+                //InputStream 'content' has been fully read to write to the temp file; so we need to use a new InputStream
+                //An alternative is to check for content.markSupported() and reset() however since we have a temporary
+                //file we may as well use it!
+                tempFIS = new FileInputStream(tempFile);
+                IOUtils.copy(tempFIS,
+                             outputStream);
+                outputStream.flush();
             }
-            outputStream = ioService.newOutputStream( nioPath,
-                                                      commentedOptionFactory.makeCommentedOption( comment,
-                                                                                                  sessionInfo.getIdentity(),
-                                                                                                  sessionInfo ) );
-
-            //InputStream 'content' has been fully read to write to the temp file; so we need to use a new InputStream
-            //An alternative is to check for content.markSupported() and reset() however since we have a temporary
-            //file we may as well use it!
-            tempFIS = new FileInputStream( tempFile );
-            IOUtils.copy( tempFIS,
-                          outputStream );
-            outputStream.flush();
 
             //Read Path to ensure attributes have been set
-            final Path newPath = Paths.convert( nioPath );
+            final Path newPath = Paths.convert(nioPath);
 
             return newPath;
-        } catch ( Exception e ) {
-            throw ExceptionUtilities.handleException( e );
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
         } finally {
+            ioService.endBatch();
```

Comment:
We are not closing the streams according recommendation. My comment is not about complete refactoring as it would be not trivial and probably risk for this old component. Could we at least simplify the finally block for a simpler one? It is still not correct but a little bit more readable. What do you think?
```
 finally {
            ioService.endBatch();
            try {
                if (content != null) {
                    content.close();
                }
                if (tempFIS != null) {
                    tempFIS.close();
                }
                if (tempFOS != null) {
                    tempFOS.close();
                }
                if (outputStream != null) {
                    outputStream.close();
                }
            } catch (IOException e) {
                throw ExceptionUtilities.handleException(e);
            }
        }
```

Reply:
But what if closing `content` fails, then the rest might be left open?

Reply:
Since this is Java backend code; why not use `try(...) {...}` (i.e. try-with-resources) and have the JRE close everything?

Reply:
I guess that would work and is worth a try.

Reply:
@jomarko I tried what @manstis proposed and it looks pretty good. Thank you for that.

Reply:
@manstis I trye locally `with-resources` but didn't liked my result. Thank you @Rikkola with your proper use of that.